tests/preprocessing/test_staypoints.py

```diff
@@ -454,6 +454,25 @@ def test_method_error(self, example_staypoints):
             example_staypoints.as_staypoints.generate_locations(method="unknown")
             assert error_msg == str(e.value)
 
+    def test_activity_flag(self, example_staypoints):
+        """Test if only activity staypoints are used if flag is set."""
+        # take out staypoint 6 that should have been merged with 2, 15
+        sp = example_staypoints
+        data = [True, True, True, True, False, True, True, True]
+        idx = [1, 2, 3, 5, 6, 7, 15, 80]
+        activities = pd.Series(data, index=idx)
+        sp["activity"] = activities
+        sp, _ = sp.as_staypoints.generate_locations(
+            method="dbscan",
+            epsilon=10,
+            num_samples=2,
+            distance_metric="haversine",
+            agg_level="user",
+            activities_only=True,
+        )
+        assert sp.loc[1, "location_id"] == sp.loc[15, "location_id"]
+        assert sp.loc[2, "location_id"] is pd.NA
```

Comment:
Would be great to check if the type of `location_id` is the same for `activity sp not assigned to locations` and `non-activity sp`.

Reply:
I just found [All NA-like values are replaced with pandas.NA](https://pandas.pydata.org/docs/user_guide/integer_na.html), as we explicitly transformed the datatype to `Int64` in Line 190. 

So this issue is solved.